zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/client/DedicatedZkClient.java

```diff
@@ -56,32 +59,42 @@
   private final ZkClient _rawZkClient;
   private final MetadataStoreRoutingData _metadataStoreRoutingData;
   private final String _zkRealmShardingKey;
-  private final String _zkRealmAddress;
 
-  // TODO: Remove MetadataStoreRoutingData from constructor
+  /**
+   * DedicatedZkClient connects to a single ZK realm and supports full ZkClient functionalities
+   * such as CRUD, change callback, and ephemeral operations for a single ZkRealmShardingKey.
+   * @param connectionConfig
+   * @param clientConfig
+   * @throws IOException
+   * @throws InvalidRoutingDataException
+   */
   public DedicatedZkClient(RealmAwareZkClient.RealmAwareZkConnectionConfig connectionConfig,
-      RealmAwareZkClient.RealmAwareZkClientConfig clientConfig,
-      MetadataStoreRoutingData metadataStoreRoutingData) {
-
+      RealmAwareZkClient.RealmAwareZkClientConfig clientConfig)
+      throws IOException, InvalidRoutingDataException {
     if (connectionConfig == null) {
       throw new IllegalArgumentException("RealmAwareZkConnectionConfig cannot be null!");
     }
-    _zkRealmShardingKey = connectionConfig.getZkRealmShardingKey();
+    if (clientConfig == null) {
+      throw new IllegalArgumentException("RealmAwareZkConnectionConfig cannot be null!");
```

Comment:
Typo in exception message. Should be RealmAwareZkClientConfig

Reply:
Good catch. Fixed